server/service/teams.go

```diff
@@ -176,6 +179,18 @@ type applyTeamSpecsRequest struct {
 	Specs  []*fleet.TeamSpec `json:"specs"`
 }
 
+func (req *applyTeamSpecsRequest) DecodeBody(ctx context.Context, r io.Reader) error {
+	if err := fleet.JSONStrictDecode(r, req); err != nil {
+		err = fleet.NewUserMessageError(err, http.StatusBadRequest)
+		if !req.Force || !fleet.IsJSONUnknownFieldError(err) {
+			// only unknown field errors can be forced at this point (other errors
+			// can be forced later, after agent options' validations)
+			return ctxerr.Wrap(ctx, err, "strict decode team specs")
```

Comment:
I thought `Force` only changes the behaviour of decoding the agent options. According to the API docs

> Force apply the options even if there are validation errors

However, setting it to `true` changes how the whole request is decoded, not just the agent options. Do the docs need to be updated? Is there a way to do strict decoding only on the agent options?

Reply:
It can be used to force-ignore unknown keys eg. at the top-level or somewhere other than in the agent options (in addition to inside the agent options). I'll update the docs' description.